File: Amy/pub_sub_pack_cpp/src/my_subscriber.cpp
//Copyright 2026 Amy Kibara
/* 
 * @author Amy Kibara
 * @file my_subscriber.cpp
 * @brief This ros2 node subscribes to a topic and prints out the messages it receives
 * @version 0.1
 * @date 2026.06.17
 * ------
 * Subscription topics:
 *  String message
 *  /greetings_cpp - std_msgs/String
 * -----
 * Publishing topics:
 *  None
 *
*/

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using std::placeholders::_1; //placeholder representing the first argument passed to the function (the message)

class Mincppsub : public rclcpp::Node
{
    public:
        Mincppsub():
         Node("min_cpp_sub")
        {
        //create var to store the sub, give it message type, topic name, queue size, fxn that's bound to it and will be executed every time a message is published on topic
            subscriber_1 = 
            create_subscription<std_msgs::msg::String>("/greetings_cpp", 10, std::bind(&Mincppsub::topicCallback, this, _1));
        }
        //callback fxn
        void topicCallback(const std_msgs::msg::String & msg) const
        {
            //every time message received, prints it out
            RCLCPP_INFO_STREAM(get_logger(), "I heard : " << msg.data);
        }
    private:
        //member vars
        rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscriber_1;
};

//initialize system, create node instance, shutdown ros2 when program end
int main(int argc, char * argv[]){
    rclcpp::init(argc, argv);
    auto min_cpp_sub_node = std::make_shared<Mincppsub>();
    rclcpp::spin(min_cpp_sub_node);
    rclcpp::shutdown();
    return 0;
}

File: play_ground/Amy/amy_ws/pub_sub_pack_py/pub_sub_pack_py/my_subscriber.py
# Copyright 2026 Amy Kibara
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

#!/usr/bin/env python3
"""
Description:
    This ROS2 node subscribes to 'Hello there!' messages
-----
Publishing topics
    None
-----
Subscription topics
    The channel containing the 'Hello there!' messages
    /greetings_py - std_msgs/String
-----
Author: Amy Kibara
Date : 17/6/2026
"""

import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class Minpysub(Node):
    def __init__(self):
        super().__init__('minpysub')
        # [msg_type, topic name, callback fxn, queue size]
        self.subscriber_1 = self.create_subscription(
            String, '/greetings_py', self.listener_callback, 10)

    def listener_callback(self, msg):
        # prints out every time it receives a message
        self.get_logger().info(f'I heard "{msg.data}"')


def main(args=None):
    rclpy.init(args=args)  # initializes ros2 comms
    myminsub = Minpysub()
    rclpy.spin(myminsub)  # keeps node running
    myminsub.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()  # executes main if script run directly